tests/test_configurations/test_base.py

```diff
@@ -1123,3 +1130,125 @@ def test_component_id_from_json():
+            r"jquery-1.2.1.js",
+            r"jquery-1.2.1.min.js",
+        ]
+
+    def yaml_config_dump_load_equal(self, data: dict) -> bool:
+        """Check whether config is the same after yaml dump and load"""
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_file = Path(tmp_dir) / "tmp_file"
+            with open(tmp_file, "w") as stream:
+                yaml_dump(data, stream)
+            with open(tmp_file, "r") as stream:
+                reconstituted_data = yaml_load(stream)
+            return data == reconstituted_data
+
+    def get_hexdigest_from_config_json(
+        self,
+        config: Union[
+            SkillComponentConfiguration, SkillConfig, AgentConfig, ContractConfig
+        ],
+    ) -> str:
+        """Get hexdigest from the json serialized configuration"""
+
+        # this is tested because we assume config.json is deterministic
+        return hashlib.sha512(json.dumps(config.json).encode("utf-8")).hexdigest()
+
+    def test_filepath_ordering(self) -> None:
+        """Test that filepath ordering remains equivalent when converting to POSIX"""
+
+        # https://learn.microsoft.com/en-us/dotnet/standard/io/file-path-formats
+
+        def sorted_indices(li: List[str]) -> List[int]:
+            return sorted(range(len(li)), key=lambda i: li[i])
+
+        # since automagically converted to POSIX compliant-format when on Linux,
+        # hard-coded the expected posix order to ensure consistency across OS.
+        expected = [3, 2, 5, 0, 4, 1, 11, 9, 7, 10, 8, 6, 14, 12, 13, 15, 16, 17]
+        posix_paths = [Path(p).as_posix() for p in self.raw_paths]
+        assert sorted_indices(self.raw_paths) == sorted_indices(posix_paths)
+        assert sorted_indices(self.raw_paths) == expected
+
+    def test_skill_component_configuration_serialization(self) -> None:
+        """Test SkillComponentConfiguration serialization"""
+
+        config = SkillComponentConfiguration(
+            class_name="class_name", file_path=self.raw_paths[0]
+        )
+        expected = "813554f9b5a750bad28bfd7005705bfc001003ac2ed2970d9030cf1b2d2ec37db69390cfc810f141e228bd3232238a9462259c2df6a3f87631606f47d0183cf8"
+        assert self.yaml_config_dump_load_equal(config.json)
+        assert self.get_hexdigest_from_config_json(config) == expected
+
+    def test_skill_configuration_serialization(self) -> None:
+        """Test SkillConfig serialization"""
+
+        name, author = "name", "author"
+        config = SkillConfig(
+            name=name,
+            author=author,
+            build_entrypoint=self.raw_paths[0],
+            build_directory=self.raw_paths[0],
+        )
+        expected = "6de1f8810c55f1ae3d9cea3c58bf41e45049ca8a8078540d5550246a0b3783dbed60672f559a38d51cf802416b8f49b7b649e367dc840ed01cda38ef19a57182"
+        assert self.yaml_config_dump_load_equal(config.json)
+        assert self.get_hexdigest_from_config_json(config) == expected
```

Comment:
You should also add more explicit check for serialised path to make sure they're dumped as posix paths

Reply:
I can recursively check the values in the dictionary (json) and if string type check that converting to posix format they remain constant.